### DISCUSSION TITLE
'_app' ignored when using  '@next/mdx'?

### DISCUSSION THREAD

**Discussion by tombuckley91** [vercel/next.js#21780] [Help]
Forgive my naivety if I'm completely approaching this from the wrong place, but I'm unsure as to why my _app file is being ignored when working with MDX. I'm currently working on a docs site where I want to be able to write something like the following in an  MDX file:

```
```graphql
# { "graphiql": true, "schema": "Users" }
{
  users {
    name
   }
}
```

And this will ultimately render an 'inline graphiql' instance, as demonstrated here [GitHub GQL Example](https://graphql.org/learn/schema/).

During my research into this, I bumped into the following article [NextJS MD Blog](https://nextjs.org/blog/markdown), which demonstrated the following pattern at the PAGE level: 

```
// pages/index.js

import { MDXProvider } from '@mdx-js/react'
import Image from 'next/image'
import { Heading, Text, Pre, Code, Table } from '../components'

const components = {
  img: Image,
  h1: Heading.H1,
  h2: Heading.H2,
  p: Text,
  code: Pre,
  inlineCode: Code
}

export default function Post(props) {
  return (
    <MDXProvider components={components}>
      <main {...props} />
    </MDXProvider>
  )
}
```

My intention was to implement this pattern at the ```_app``` level, meaning that the individual pages can be written in MD and need no extra boiler plate at the page level. To investigate what kind of information I would have available in ```_app``` I put the following file together:

```
// /pages/_app.tsx

import { MDXProvider } from "@mdx-js/react";

import type { AppProps } from "next/app";

function MyApp({ Component, pageProps }: AppProps) {
  const components = {
      code: function () {
        const args = Array.from(arguments);
        console.log("I am the provided args: ", args);

        return <code>Hello World</code>;
    },
  };

  console.log("MY APP IS EVEN BEING CALLED!");

  return (
    <MDXProvider components={components}>
      <Component {...pageProps} />
    </MDXProvider>
  );
}

export default MyApp;

``` 

During this work, I realised that my ```_app``` was not even being invoked, which I have no understanding as to why that would be the case. 

Effectively, all I want to be able to do is to override the default  HTML elements that would usually be created by MDX with custom ones, but to only have to define these in a single place. Any help with this would be greatly appreciated, and again, I apologise in advance if I'm completely attacking this in the wrong way. 

Thanks,
Tom

Side note - I am perfectly aware that I could just import a component and use that in my MDX, that is not an option, the MD should be capable of being created by non-technical members.

**Comment by tombuckley91** [comment]
So just to give a little more insight into this, this problem only exists when the repository uses TypeScript, recreating this in JS correctly invokes _app.

**Comment by twavv** [comment]
I just ran into this. The issue was the configuration that the docs suggest is

```
// next.config.js
const withMDX = require('@next/mdx')()
module.exports = withMDX({
  pageExtensions: ['js', 'jsx', 'mdx'],
})
```
which excludes `tsx`.

The correct config for me was:
```js
const withMDX = require('@next/mdx')({
  options: {
    remarkPlugins: [],
    rehypePlugins: [],
  },
})
module.exports = withMDX({
  pageExtensions: ['js', 'jsx', "tsx", 'mdx'],
})
```
